cluster-autoscaler/core/scale_down.go

```diff
@@ -678,19 +690,20 @@ func (sd *ScaleDown) markSimulationError(simulatorErr errors.AutoscalerError,
 // chooseCandidates splits nodes into current candidates for scale-down and the
 // rest. Current candidates are unneeded nodes from the previous run that are
 // still in the nodes list.
-func (sd *ScaleDown) chooseCandidates(nodes []*apiv1.Node) ([]*apiv1.Node, []*apiv1.Node) {
+func (sd *ScaleDown) chooseCandidates(nodes []*schedulernodeinfo.NodeInfo) ([]*schedulernodeinfo.NodeInfo, []*schedulernodeinfo.NodeInfo) {
```

Comment:
See my comment in simulator/cluster.go: there is really no need for this to be NIs. All we care about are node names.

Reply:
Passing around node names means checking errors every time we retrieve the node from the snapshot.

I'm leaning towards using node infos retrieved once from snapshot, but for now implemented it as requested.